CrazyCanvas/Source/ECS/Systems/GUI/HUDSystem.cpp

```diff
@@ -89,39 +89,43 @@ void HUDSystem::FixedTick(Timestamp delta)
 
 bool HUDSystem::OnWeaponFired(const WeaponFiredEvent& event)
 {
-	ECSCore* pECS = ECSCore::GetInstance();
-	const ComponentArray<WeaponComponent>* pWeaponComponents = pECS->GetComponentArray<WeaponComponent>();
-
-	for (Entity playerWeapon : m_WeaponEntities)
+	if (!MultiplayerUtils::IsServer())
 	{
-		const WeaponComponent& weaponComponent = pWeaponComponents->GetConstData(playerWeapon);
+		ECSCore* pECS = ECSCore::GetInstance();
+		const ComponentArray<WeaponComponent>* pWeaponComponents = pECS->GetComponentArray<WeaponComponent>();
+		const ComponentArray<PlayerLocalComponent>* pPlayerLocalComponents = pECS->GetComponentArray<PlayerLocalComponent>();
 
-		if (weaponComponent.WeaponOwner == event.WeaponOwnerEntity && m_HUDGUI)
+		for (Entity playerWeapon : m_WeaponEntities)
 		{
-			m_HUDGUI->UpdateAmmo(weaponComponent.WeaponTypeAmmo, event.AmmoType);
+			const WeaponComponent& weaponComponent = pWeaponComponents->GetConstData(playerWeapon);
+
+			if (pPlayerLocalComponents->HasComponent(weaponComponent.WeaponOwner) && m_HUDGUI)
```

Comment:
Is there a reason for not subscribing directly to PlayerLocalComponent when you subscribe to the weapon-component? This should allow you to skip this check.